app/src/main/java/com/example/leagueoflegends/view/MainActivity.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.leagueoflegends.view
+
+import android.annotation.SuppressLint
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import com.example.leagueoflegends.R
+import com.example.leagueoflegends.contract.BaseContract
+import com.example.leagueoflegends.misc.MainAdapter
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main.*
+import org.koin.android.ext.android.inject
+import java.lang.Exception
+
+class MainActivity : AppCompatActivity(), BaseContract.BaseView {
+
+    private val presenter: BaseContract.BasePresenter by inject()
+
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        presenter.onAttach(this)
+        displayChampions()
+    }
+
+    @SuppressLint("CheckResult")
+    private fun displayChampions() {
+        val adapter = MainAdapter()
+        presenter.getChampions()
```

Comment:
this API call shouldn't be in the View. This is the responsibility of the presenter layer. and I also wouldn't bring RX here. Presenter calls getChampions, does this rx thing inside, and then gives the data to a view.display(data) function. it goes around, otherwise, it's not MVP pattern

Reply:
thats why you have those view functions in the contract, so you can call them from the presenter